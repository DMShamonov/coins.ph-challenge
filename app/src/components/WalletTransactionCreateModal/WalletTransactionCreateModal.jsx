// @flow

import * as React from 'react';

import Modal from 'components/Modal';
import WalletTransactionForm from 'components/WalletTransactionForm';

import type { PropsType } from './types.js.flow';

export default function WalletTransactionCreateModal({
  isOpen,
  walletId,
  onClose,
}: PropsType): React.Node {
  return (
    <Modal isOpen={isOpen} contentLabel="Create wallet transaction" onClose={onClose}>
      <WalletTransactionForm walletId={walletId} />
    </Modal>
  );
}
