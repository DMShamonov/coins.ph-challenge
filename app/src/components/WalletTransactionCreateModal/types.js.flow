// @flow

export type PropsType = {|
  isOpen: boolean,
  walletId: string,
  onClose: () => void,
|};
