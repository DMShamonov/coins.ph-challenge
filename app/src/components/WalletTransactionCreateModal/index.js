import { connect } from 'react-redux';

import MODAL_ACTION_TYPES from 'redux/actionTypes/modal';
import MODALS from 'constants/modals';

import WalletTransactionCreateModal from './WalletTransactionCreateModal';

function mapStateToProps({ ui: { modal } }, { walletId }) {
  return {
    isOpen: modal === MODALS.CREATE_WALLET_TRANSACTION,
    walletId,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    onClose() {
      dispatch({ type: MODAL_ACTION_TYPES.ON_HIDE });
    },
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(WalletTransactionCreateModal);
