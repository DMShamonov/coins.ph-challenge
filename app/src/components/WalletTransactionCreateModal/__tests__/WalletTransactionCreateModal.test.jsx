import React from 'react';
import { shallow } from 'enzyme';

import WalletTransactionCreateModal from '../WalletTransactionCreateModal';

describe('`WalletTransactionCreateModal` component should render correctly', () => {
  test('with default props', () => {
    expect(shallow((
      <WalletTransactionCreateModal
        isOpen={false}
        walletId="__WALLET_ID__"
        onClose={jest.fn()}
      />
    ))).toMatchSnapshot();
  });

  test('when `isOpen` prop is `true`', () => {
    expect(shallow((
      <WalletTransactionCreateModal
        isOpen
        walletId="__WALLET_ID__"
        onClose={jest.fn()}
      />
    ))).toMatchSnapshot();
  });
});
