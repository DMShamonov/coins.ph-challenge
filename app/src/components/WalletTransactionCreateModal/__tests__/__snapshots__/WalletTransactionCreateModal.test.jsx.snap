// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`WalletTransactionCreateModal\` component should render correctly when \`isOpen\` prop is \`true\` 1`] = `
<Modal
  contentLabel="Create wallet transaction"
  isOpen={true}
  onClose={[MockFunction]}
>
  <Connect(WalletTransactionForm)
    walletId="__WALLET_ID__"
  />
</Modal>
`;

exports[`\`WalletTransactionCreateModal\` component should render correctly with default props 1`] = `
<Modal
  contentLabel="Create wallet transaction"
  isOpen={false}
  onClose={[MockFunction]}
>
  <Connect(WalletTransactionForm)
    walletId="__WALLET_ID__"
  />
</Modal>
`;
